src/components/GestureButtons.tsx

```diff
@@ -139,28 +182,25 @@ export class RectButton extends React.Component {
   }
 }
 
-export class BorderlessButton extends React.Component {
-  static propTypes = {
-    ...BaseButton.propTypes,
-    borderless: PropTypes.bool,
-  };
-
+export class BorderlessButton extends React.Component<BorderlessButtonProperties> {
   static defaultProps = {
     activeOpacity: 0.3,
     borderless: true,
   };
 
-  constructor(props) {
+  constructor(props: BorderlessButtonProperties) {
     super(props);
-    this._opacity = new Animated.Value(1);
+    this.opacity = new Animated.Value(1);
   }
 
-  _onActiveStateChange = (active) => {
+  private opacity: Animated.Value;
+
+  private onActiveStateChange = (active: boolean) => {
     if (Platform.OS !== 'android') {
-      this._opacity.setValue(active ? this.props.activeOpacity : 1);
+      this.opacity.setValue(active ? this.props.activeOpacity! : 1);
```

Comment:
Not null type assertion.

Reply:
It is inevitable in class component with `defaultProps`, which take care of always providing the value.